source/nodejs/adaptivecards-designer/src/app.ts

```diff
@@ -237,65 +264,56 @@ class DesignerApp {
         if (this.paletteHostElement) {
             this.paletteHostElement.innerHTML = "";
 
-            var sortedRegisteredTypes: Array<Adaptive.ITypeRegistration<Adaptive.CardElement>> = [];
-
-            for (var i = 0; i < Adaptive.AdaptiveCard.elementTypeRegistry.getItemCount(); i++) {
-                sortedRegisteredTypes.push(Adaptive.AdaptiveCard.elementTypeRegistry.getItemAt(i));
+            let sortedRegisteredTypes = {};
+
+            const categoriesMap = {
+                cardElements: {
+                    title: "Card Elements",
+                    items: ["TextBlock", "Image"]
+                },
+                container: {
+                    title: "Container",
+                    items: ["Container", "ColumnSet", "Column", "FactSet", "Fact", "ImageSet"]
+                },
+                actions: {
+                    title: "Actions",
+                    items: ["Action.OpenUrl", "Action.Submit", "Action.ShowCard"]
+                },
+                inputs: {
+                    title: "Inputs",
+                    items: ["Input.Text", "Input.Number", "Input.Date", "Input.Time", "Input.Toggle", "Input.ChoiceSet", "Input.Choice"]
+                }
             }
 
-            sortedRegisteredTypes.sort(
-                (a, b) => {
-                    if (a.typeName < b.typeName) {
-                        return -1
-                    }
-                    else {
-                        return 1;
+            let categoriesMapKeys = Object.keys(categoriesMap);
```

Comment:
Please rework this with category being a property of ITypeRegistration.

Reply:
P0: category needs to be passed in constructor for peer registration - additional map as created here needs go